### PR TITLE
chore: Update to JDK 17

### PR DESCRIPTION
**Pull request by mshabarov**
MERGE, DO NOT SQUASH

### CODE REVIEW

File: scripts/computeMatrix.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const fs = require("fs");
 
 /****************** START CONFIG */
 // Do not run the following modules except 'flow-tests' that is handled separatelly in this script
-const globalExclusions = ['flow-tests/servlet-containers/tomcat9', 'flow-tests/servlet-containers/tomcat85'];
+const globalExclusions = ['flow-tests/servlet-containers/tomcat10', 'flow-tests/servlet-containers/tomcat9', 'flow-tests/servlet-containers/tomcat85'];
```
Comment (Artur-) [inline scripts/computeMatrix.js:6]: This seems unrelated

File: pom.xml [hunk 2]
```diff
@@ -677,5 +677,18 @@
                 <sonar.issuesReport.html.enable/>
             </properties>
         </profile>
+	<!-- TEMPORARY override apicmp profile in vaadin-parent until we release it with this change -->
```
Comment (Artur-) [inline pom.xml:680]: Isn't this released already in https://repo1.maven.org/maven2/com/vaadin/vaadin-parent/2.1.0/vaadin-parent-2.1.0.pom ?

**Review by Artur- — DISMISSED** [review]

**Comment by Artur-** [comment]
Should flow-plugins/flow-gradle-plugin/gradle/wrapper/gradle-wrapper.properties also be updated here?

**Comment by mshabarov** [comment]
> Should flow-plugins/flow-gradle-plugin/gradle/wrapper/gradle-wrapper.properties also be updated here?

Yes, I can include it as well.

**Comment by Artur-** [comment]
The wrapper binary files also?

**Review by Artur- — APPROVED** [review]
